### PR TITLE
[CBRD-20555] Optimizes heap insertion by skipping pack, unpack for unmodified fields

### PR DESCRIPTION
**Pull request by alexstanro**
http://jira.cubrid.org/browse/CBRD-20555

### CODE REVIEW

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/heap_file.c (1 incorrect indentation(s) found)
```
 20872       bool use_optimization;
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

File: src/storage/heap_file.c [hunk 1]
```diff
@@ -19901,47 +19903,171 @@ heap_build_forwarding_recdes (RECDES * recdes_p, INT16 rec_type, OID * forward_o
...
+  if (use_optimization)
+    {
+      /* 
+       * Most common case. Since is UPDATE_INPLACE_NONE, the header does not have DELID and PREV VERSION.
+       * Optimize header adjustment.
+       */
+      assert (!(mvcc_flags & OR_MVCC_FLAG_VALID_DELID));
+      assert (!(mvcc_flags & OR_MVCC_FLAG_VALID_PREV_VERSION));
+      mvcc_id = logtb_get_current_mvccid (thread_p);
+
```
Comment (eseokoh) [inline src/storage/heap_file.c]: I'd like to propose introducing a pointer variable, `existing_data_p` and renaming `recdes_data` as `new_ins_mvccid_pos_p`.

File: src/storage/heap_file.c [hunk 2]
```diff
@@ -19901,47 +19903,171 @@ heap_build_forwarding_recdes (RECDES * recdes_p, INT16 rec_type, OID * forward_o
...
+      recdes_data = start_p = insert_context->recdes_p->data;
+      /* Skip bytes up to insid_offset */
+      recdes_data += OR_MVCC_INSERT_ID_OFFSET;
```
Comment (eseokoh) [inline src/storage/heap_file.c]: Rewrite the lines as:
```C
      start_p = insert_context->recdes_p->data;
      /* Skip bytes up to insid_offset */
      new_ins_mvccid_pos_p = start_p + OR_MVCC_INSERT_ID_OFFSET;
      existing_data_p = new_ins_mvccid_pos_p;
```

File: src/storage/heap_file.c [hunk 3]
```diff
@@ -19901,47 +19903,171 @@ heap_build_forwarding_recdes (RECDES * recdes_p, INT16 rec_type, OID * forward_o
...
+      if (!(mvcc_flags & OR_MVCC_FLAG_VALID_INSID))
+	{
+	  /* Sets MVCC INSID flag, overwrite first four bytes. */
+	  repid_and_flag_bits |= (OR_MVCC_FLAG_VALID_INSID << OR_MVCC_FLAG_SHIFT_BITS);
+	  OR_PUT_INT (start_p, repid_and_flag_bits);
+
+	  /* Move the record data before inserting INSID */
+	  assert (insert_context->recdes_p->area_size >= insert_context->recdes_p->length + OR_MVCCID_SIZE);
+	  memmove (recdes_data + OR_MVCCID_SIZE, recdes_data,
+		   insert_context->recdes_p->length - OR_MVCC_INSERT_ID_OFFSET);
```
Comment (eseokoh) [inline src/storage/heap_file.c]:
```C
          memmove (new_ins_mvccid_pos_p + OR_MVCCID_SIZE, existing_data_p,
                   insert_context->recdes_p->length - OR_MVCC_INSERT_ID_OFFSET);
```

File: src/storage/heap_file.c [hunk 4]
```diff
@@ -19901,47 +19903,171 @@ heap_build_forwarding_recdes (RECDES * recdes_p, INT16 rec_type, OID * forward_o
...
+	  insert_context->recdes_p->length += OR_MVCCID_SIZE;
+	}
+
+      /* Sets the MVCC INSID */
+      OR_PUT_BIGINT (recdes_data, &mvcc_id);
```
Comment (eseokoh) [inline src/storage/heap_file.c]:
```C
      OR_PUT_BIGINT (new_ins_mvccid_pos_p, &mvcc_id);
```

File: src/storage/heap_file.c [hunk 5]
```diff
@@ -19901,47 +19903,171 @@ heap_build_forwarding_recdes (RECDES * recdes_p, INT16 rec_type, OID * forward_o
  * heap_insert_adjust_recdes_header () - adjust record header for insert
  *                                       operation
  *   thread_p(in): thread entry
- *   context(in): operation context
- *   recdes_p(in): record descriptor to adjust
- *   is_mvcc_op(in): specifies type of operation (MVCC/non-MVCC)
+ *   insert_context(in): insert context
```
Comment (eseokoh) [inline src/storage/heap_file.c]:
```C
+ *   insert_context(in/out): insert context
```

File: src/storage/heap_file.c [hunk 6]
```diff
@@ -19901,47 +19903,171 @@ heap_build_forwarding_recdes (RECDES * recdes_p, INT16 rec_type, OID * forward_o
...
+  /* all ok */
+  return NO_ERROR;
+}
 
-  if (insert_from_reorganize == true && MVCC_IS_HEADER_DELID_VALID (&mvcc_rec_header))
+/*
+ * heap_update_adjust_recdes_header () - adjust record header for update
+ *                                       operation
+ *   thread_p(in): thread entry
+ *   update_context(in): update context
```
Comment (eseokoh) [inline src/storage/heap_file.c]:
```c
+ *   update_context(in/out): update context
```

**Pull request merged**